tools/make_patch.c

```diff
@@ -10,40 +10,67 @@
 
 struct Buffer {
 	size_t size;
-	char buffer[];
+	char data[];
 };
 
 struct Patch {
 	unsigned int offset;
 	unsigned int size;
 };
 
+struct Patches {
+	size_t size;
+	size_t capacity;
+	struct Patch *data;
+};
+
 struct Symbol {
 	struct Symbol *next;
 	int address;
 	unsigned int offset;
 	char name[];
 };
 
-void *create_buffer(void) {
-	struct Buffer *buffer = xmalloc(sizeof(struct Buffer) + 0x100);
+char *create_buffer(void) {
+	struct Buffer *buffer = xmalloc(sizeof(*buffer) + 0x100);
 	buffer->size = 0x100;
-	return buffer->buffer;
+	return buffer->data;
 }
 
-void free_buffer(void *buf) {
-	struct Buffer *buffer = (struct Buffer *)((char *)buf - sizeof(struct Buffer));
+void free_buffer(char *data) {
+	struct Buffer *buffer = (struct Buffer *)(data - sizeof(*buffer));
```

Comment:
`clang`'s `-Wcast-align` warns here about casting from align 1 to align 8. Of course, this function has the precondition that `char *data` isn't any old `char *` pointer, but one that's derived from a `struct Buffer *`; but I'd like a solution that the compiler's type system is capable of verifying. :P

Reply:
That warning is bogus in this context...

Reply:
Right, I was just complaining about a code style of "pointer manipulation too weird for the compiler to ensure things are aligned properly so you have to do that yourself".